Source/Tools/FEXServer/CoreDumpService.cpp

```diff
@@ -0,0 +1,365 @@
+        case FEXServerClient::CoreDump::PacketTypes::HOST_CONTEXT: {
+          FEXServerClient::CoreDump::PacketHostContext *Req = reinterpret_cast<FEXServerClient::CoreDump::PacketHostContext *>(&Data[CurrentOffset]);
+          HostContextData = *Req;
+          // TODO: Create host context unwinder.
+          CurrentOffset += sizeof(*Req);
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::GUEST_CONTEXT: {
+          FEXServerClient::CoreDump::PacketGuestContext *Req = reinterpret_cast<FEXServerClient::CoreDump::PacketGuestContext *>(&Data[CurrentOffset]);
+          Desc.GuestArch = Req->GuestArch;
+          // TODO: Create guest context unwinder.
+          CurrentOffset += sizeof(*Req);
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::CONTEXT_UNWIND: {
+          // Print the header to the backtrace.
+          BacktraceHeader();
+          // TODO: Unwind the host and guest contexts that were received.
+          // Shutdown the client once FEXServer has unwound.
+          FEXServerClient::CoreDump::SendShutdownPacket(ServerSocket);
+
+          // Wait a moment for the client to sanely clean-up.
+          // Then just leave if it hung regardless.
+          std::this_thread::sleep_for(std::chrono::seconds(1000));
+          ShouldShutdown = true;
+          CurrentOffset += sizeof(FEXServerClient::CoreDump::PacketHeader);
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::GET_JIT_REGIONS: {
+          FEXServerClient::CoreDump::PacketGetJITRegions *Req = reinterpret_cast<FEXServerClient::CoreDump::PacketGetJITRegions *>(&Data[CurrentOffset]);
+          ThreadDispatcher = Req->Dispatcher;
+          for (size_t i = 0; i < Req->NumJITRegions; ++i) {
+            ThreadJITRegions.emplace_back(Req->JITRegions[i]);
+          }
+          CurrentOffset += sizeof(*Req) + Req->NumJITRegions * sizeof(FEXCore::Context::Context::JITRegionPairs);
+          break;
+        }
+        default:
+          // Something sent us an invalid packet. To ensure we don't spin infinitely, consume all the data.
+          LogMan::Msg::EFmt("[CoreDumpClass] InvalidPacket size received 0x{:x} bytes", CurrentRead - CurrentOffset);
+          CurrentOffset = CurrentRead;
+          break;
+      }
+    }
+  }
+
+  void CoreDumpClass::ExecutionFunc() {
+    auto LastDataTime = std::chrono::system_clock::now();
+
+    while (!ShouldShutdown) {
+      struct timespec ts{};
+      ts.tv_sec = RequestTimeout;
+
+      if (PollFDs.empty()) {
+        ShouldShutdown = true;
+      }
+      else {
+        int Result = ppoll(&PollFDs.at(0), PollFDs.size(), &ts, nullptr);
+        if (Result > 0) {
+          // Walk the FDs and see if we got any results
+          for (auto it = PollFDs.begin(); it != PollFDs.end(); ) {
+            auto &Event = *it;
+            bool Erase{};
+
+            if (Event.revents != 0) {
+              if (Event.fd == ServerSocket) {
```

Comment:
You're right, no other FDs get added to the vector. I had some plans for multiple sockets there but it ended up be unnecessary.
Refactored to remove the vector since it only ever polls a single pollfd per CoreDumpClass.